doc/platform_api/CMIS_Diagnostic_Monitoring_Overview_in_SONiC.md

```diff
@@ -0,0 +1,2375 @@
+    - If the flag to update all diagnostic information is set, it reads the diagnostic information for the current port.
+    - After processing all ports, it resets the flag and updates the `expired_time` to the current time plus the `dom_info_update_periodic_secs` interval.
+
+The following steps are performed to update all diagnostic information for a port:
+
+1. Ensure DOM monitoring is enabled for the port (in case of breakout port, port refers to the first subport of the breakout port group). If DOM monitoring is disabled, skip updating the diagnostic information for the port.
+2. Read the transceiver firmware information from the module and update the `TRANSCEIVER_FIRMWARE_INFO` table.
+3. Read the transceiver DOM sensor data from the module and update the `TRANSCEIVER_DOM_SENSOR` table.
+4. Read the transceiver DOM flag data from the module, record the timestamp, and update the `TRANSCEIVER_DOM_FLAG` table (update the `TRANSCEIVER_DOM_FLAG_CHANGE_COUNT`, `TRANSCEIVER_DOM_FLAG_SET_TIME`, and `TRANSCEIVER_DOM_FLAG_CLEAR_TIME` tables as well).
+5. Read the transceiver status data from the module and update the `TRANSCEIVER_STATUS` table.
+6. Read the transceiver status flag data from the module, record the timestamp, and update the `TRANSCEIVER_STATUS_FLAG` table (update the `TRANSCEIVER_STATUS_FLAG_CHANGE_COUNT`, `TRANSCEIVER_STATUS_FLAG_SET_TIME`, and `TRANSCEIVER_STATUS_FLAG_CLEAR_TIME` tables as well).
+7. If the transceiver supports VDM monitoring, perform the following steps:
+    1. Freeze the statistics by calling the CMIS API (`freeze_vdm_stats`) and wait for `FreezeDone` by calling `get_vdm_freeze_status`.
+    2. Once the statistics are frozen, read the VDM real values, flags, and PM data from the module and update the `TRANSCEIVER_VDM_REAL_VALUE`, `TRANSCEIVER_VDM_FLAG`, and `TRANSCEIVER_PM` tables respectively.
+    3. Update the VDM flag, change count, and time-related tables by comparing the current data with the previous data.
+    4. Unfreeze the statistics by calling the CMIS API (`unfreeze_vdm_stats`).
+
+Pseudo code:
+
+```python
+# Retrieve the update period interval
+dom_info_update_periodic_secs = parse_field_from_pmon_daemon_control('dom_info_update_periodic_secs')
+if dom_info_update_periodic_secs is None:
+    dom_info_update_periodic_secs = 0
+
+expired_time = time.time()
+
+while not dom_mgr.task_stopping_event.is_set():
+    if expired_time <= time.time():
+        update_all_diagnostic_info_in_db = True
+    
+    for current_pport in range(1, last_physical_port + 1):
+        if handle_port_update_event_for_all_link_changed_ports():
+            update_flag_related_tables(ports_going_through_link_change)
+        
+        if update_all_diagnostic_info_in_db:
+            read_diagnostic_info(current_pport)
+    
+    update_all_diagnostic_info_in_db = False
+    expired_time = time.time() + dom_info_update_periodic_secs
```

Comment:
shouldn't expired_time be computed by accounting for lost time in reading the dom values especially where lost time < `dom_info_update_periodic_secs`